BI624_deduper_P1.Rmd

```diff
@@ -0,0 +1,57 @@
+
+    1. Strand orientation
+    2. Known UMI
+    3. Chromsome number
+    4. left-most mapping starting position(basepair starting position)
+
+#### Developing some Pseudocode
+
+
+- begin by sorting the sam file with samtools sort. This will organize the reads by chromsome and starting position on the command line
+- initiate an empty list
+- loop through values of known UMIs and add those to the empty list above
+- initiate an empty dictionary
+- set a variable called previous chromsome to 1 so that we can compare chromosome values later
+- loop through each line of the sam file
+    - seperate the sam file into parts and save each of the following 4 components: UMI, strand value(0 or 16), chromsome and start position with soft 
+    - look at the cigar string
+    - if there has been soft clipping, then determine by how much and adjust the start position accordingly and overwrite this value
+    - check if the UMI is in the list of known UMIs, if it is then continue on
+          - check if the chromsome is equal to the previous value and if it is then continue on
+          - set current chromosome value equal to past chromosome value
+          - generate a tuple of this information
+          - check if the tuple is in the dictionary, if not then add it to the dictionary with a count of 1 and write that line to an output file * the counter here acts as a sanity check, since if the dictionary is printed and there are numbers >1, then there is a bug in the code
+          - if it isnt then move on to the next line and do nothing else to this line
+
+
```

Comment:
The script looks solid. You're employing an approach very similar to mine. (As I'm sure alot of us are).  The only thing I'd mention is you're going to have a dictionary with Tons of keys and really no values. I used the UMI's as the keys and then a list within each for your '4 components'. Just makes it a little easier to follow. The only thing is you wouldn't be able to have your sanity check =( . In any case looks great, just throwing some ideas at you!